aten/src/ATen/native/cuda/ForeachReduceOp.cu

```diff
@@ -267,7 +267,8 @@ std::vector<Tensor> foreach_tensor_norm_cuda(
   int i = 0;
   for (const auto& t : tensors) {
     if (t.numel() != 0) {
-      result.emplace_back(ret_per_tensor[i]);
+      result.emplace_back(at::zeros({}, ret_per_tensor.options())
+                              .fill_(ret_per_tensor[i].item()));
```

Comment:
nit: Can you use `at::empty` if you are going to `fill_` immediately after?

Reply:
This is already really bad performance-wise cuz we're creating diff kernels + using .item() (which could be replaced with a copy_, which is still not great).

Based on a discussion with @albanD, the best approach would be to pass a tensorlist of scalars and change the kernel to do that instead of having to slice up one tensor. Time for me to learn more CUDA 😛

Reply:
This is a possibility, so the new proposed solution may use more memory than the old state, but not compared to the current state where I create n new tensors anyway.

Reply:
Please don't do what's written here
1) there's no urgent need to match slow path, storage offset is an implementation detail that's not guaranteed to be preserved between fast and slow paths
2) this is a sync! Syncs should be avoided
3) this brings back "1 kernel per tensor" which for_each is meant to avoid.

Reply:
@awgu correct, proposed solution will use 256 or 512 (don't remember off-hand) bytes for each tensor, so slightly more memory, also, slightly more overhead for allocation requests

Reply:
(I think it's 512.) I've now implemented the new proposal, which will use slightly more memory + overhead, but running a profile reveals that runtime is not significantly different.